test/validate-pipeline.sh

```diff
@@ -0,0 +1,53 @@
+#!/bin/bash
+set -o errexit
+set -o nounset
+set -o pipefail
+
+SCRIPT_DIR=$( cd -- "$( dirname -- "${BASH_SOURCE[0]}" )" &> /dev/null && pwd )
+ROOT_DIR=$SCRIPT_DIR/..
+DEBUG_OUTPUT=${DEBUG_OUTPUT:-'/tmp/log.txt'}
+EVENT_TYPE=${EVENT_TYPE:-'push'}
+PIPELINE="${PIPELINE:-docker-build-rhtap}"
+TASK="${TASK:-update-deployment}"
+
+NAMESPACE="test-pipeline$((RANDOM % 10000000))"
+
+echo "Create a new test project"
+oc new-project "$NAMESPACE"
+
+wait_for_pipeline() {
+    local timeout_seconds=$((15 * 60))
+    if ! oc wait --for=condition=succeeded "$1" -n "$2" --timeout "${timeout_seconds}s" >"$DEBUG_OUTPUT"; then
+        echo "[ERROR] RHTAP Pipeline failed to complete successful" >&2
+        oc get "$1" -n "$2" -o yaml >"$DEBUG_OUTPUT"
+        exit 1
+    fi
+}
+
+echo "Preparing rhtap sample pipelines build resources..."
+"${ROOT_DIR}"/hack/build/prepare-build-resources.sh || status="$?" || :
+
+echo "Apply the rhtap tasks and pipelines in the test namespace $NAMESPACE..."
+oc apply -f "${ROOT_DIR}/pac/tasks"
+oc apply -f "${ROOT_DIR}/pac/pipelines"
+
+# Run the rhtap sample build pipeline
+"${ROOT_DIR}"/hack/build/run-build.sh || status="$?" || :
+
+wait_for_pipeline "pipelineruns/$PIPELINE" "$NAMESPACE"
+
+echo "Verify the deployment taskruns on push or pull_request event type..."
+if [[ "${EVENT_TYPE}" == "pull_request" ]]; then
+    taskruns=$(oc get pipelineruns/"$PIPELINE" -n "$NAMESPACE" -o jsonpath='{.status.skippedTasks[*].name}')
+    if ! grep -q "$TASK" <<< "$taskruns"; then
+        echo "Error: Gitops deployment repo gets updated on pull_request events"
+        exit 1
+    fi
+else
+    if oc get taskruns/"$PIPELINE"-"$TASK" -n "$NAMESPACE" -o jsonpath='{.status.conditions[*].reason}' != "Succeeded"; then
+        echo "Error: Gitops deployment repo not updated on push events"
```

Comment:
It would show correct message only for `update-deployment` task, but we have `$TASK` as a parameter.